### PR TITLE
[JENKINS-36197] Upgrade to plugin-pom 2.11

### PR DESCRIPTION
**Pull request by alvarolobato**
Upgraded to plugin-pom 2.11 and raised core version to 1.648 to avoid having to update from BC API 1.0 to 1.648 as requested on [JENKINS-36197](https://issues.jenkins-ci.org/browse/JENKINS-36197)

### CODE REVIEW

**Comment by alvarolobato** [comment]
@reviewbybees

File: pom.xml [hunk 1]
```diff
@@ -55,6 +56,7 @@ THE SOFTWARE.
 
     <properties>
         <powermock.version>1.6.3</powermock.version>
+        <jenkins.version>1.648</jenkins.version>
```
Comment (oleg-nenashev) [inline pom.xml]: 🐛 for bumping to a non-LTS version. Better to use 1.651.3 IMHO

File: src/main/resources/hudson/plugins/ec2/EC2SpotComputerLauncher/main.properties [hunk 2]
```diff
@@ -1,25 +1,24 @@
-<!--
-The MIT License
-
```
Comment (oleg-nenashev) [inline src/main/resources/hudson/plugins/ec2/EC2SpotComputerLauncher/main.properties]: Oh... wow. Thanks for catching it

**Comment by oleg-nenashev** [comment]
The version dep needs fix, other stuff LGTM.
Maybe we could avoid updating the core dependency, but it requires a decision by mentor.

The CI needs a more powerful machine from what I see. I'll fix it

**Comment by oleg-nenashev** [comment]
retriggering the build

**Comment by alvarolobato** [comment]
@oleg-nenashev changed the baseline

**Comment by oleg-nenashev** [comment]
🐝

**Pull request merged**